src/main/java/xolpoc/config/DeployerConfiguration.java

```diff
@@ -50,13 +67,30 @@ public ModuleDeployer moduleDeployer() {
 
 	@Bean
 	public ResourceModuleRegistry moduleRegistry() {
-		return new ResourceModuleRegistry(MODULE_HOME);
+		return new ResourceModuleRegistry(deployer.getModule());
 	}
 
 	@Bean
 	public ModuleFactory moduleFactory() {
 		return new ModuleFactory(moduleOptionsMetadataResolver());
 	}
+	
+	@Bean
+	@ConfigurationProperties("property")
+	public ModuleDeploymentProperties moduleDeploymentProperties() {
+		return new ModuleDeploymentProperties();
```

Comment:
This doesn't accept any provided property.* value anymore. I know it wasn't documented in any of the examples, but locally I was testing it with the following -D properties passed to 3 modules (partitioning from 1 source to 2 sinks):

````
source.time:
    -Doption.fixedDelay=5 -Doption.timeUnit=seconds -Dproperty.producer.next.module.count=2 -Dproperty.producer.partitionKeyExpression=payload -Dproperty.producer.partitionCount=2

sink.log.1:
    -Dxd.module=ticktock.sink.log.1 -Dserver.port=9999 -Dproperty.consumer.sequence=0 -Dproperty.consumer.partitionIndex=0 -Dproperty.consumer.count=2

sink.log.2:
    -Dxd.module=ticktock.sink.log.1 -Dserver.port=9998 -Dproperty.consumer.sequence=1 -Dproperty.consumer.partitionIndex=1 -Dproperty.consumer.count=2
````

I have a small change that re-enables support to pass those values:
https://github.com/markfisher/xolpoc/commit/3aefe1804c3118ec19d8cc74ab9eec899aad188d

However, when I run those 3 modules, the time source module throws this Exception (somehow related to the adjustments to the app context hierarchy perhaps?):

````
Caused by: java.lang.IllegalArgumentException: EvaluationContext is required
	at org.springframework.util.Assert.notNull(Assert.java:112)
	at org.springframework.expression.spel.standard.SpelExpression.getValue(SpelExpression.java:249)
	at org.springframework.xd.dirt.integration.bus.MessageBusSupport.determinePartition(MessageBusSupport.java:672)
	at org.springframework.xd.dirt.integration.redis.RedisMessageBus.access$600(RedisMessageBus.java:66)
	at org.springframework.xd.dirt.integration.redis.RedisMessageBus$SendingHandler.handleMessageInternal(RedisMessageBus.java:430)
	at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:78)
````

cc @garyrussell